### PR TITLE
[infra] Update base-builder image to support go-fuzz (#2714).

### PR DESCRIPTION
**Pull request by Dor1s**
If we agree on this, then we can update golang scripts and also document this.

In the meantime, I'll check locally that this does not break golang project. If it does, I'll just update its scripts in this PR as well.

### CODE REVIEW

File: infra/base-images/base-builder/Dockerfile [hunk 1]
```diff
@@ -27,6 +27,27 @@ RUN apt-get install -y git \
     libblocksruntime-dev \
     libc6-dev-i386
 
+# Download and install the latest stable Go.
+ADD https://storage.googleapis.com/golang/getgo/installer_linux $SRC/
+
+# $SHELL is needed for the installer.
+ENV SHELL "bash"
+RUN chmod +x $SRC/installer_linux && \
```
Comment (inferno-chromium) [inline infra/base-images/base-builder/Dockerfile]: nit: why not add SHELL="bash" when calling the command itself, this removes lines 34 and 38.

**Comment by Dor1s** [comment]
Travis is unhappy because this updates the base-builder image.

File: infra/base-images/base-builder/Dockerfile [hunk 2]
```diff
@@ -27,6 +27,27 @@ RUN apt-get install -y git \
...
+    $SRC/installer_linux && \
+    rm $SRC/installer_linux
+ENV SHELL ""
+
+# Set up Golang environment variables (copied from /root/.bash_profile).
+ENV PATH $PATH:/root/.go/bin
```
Comment (inferno-chromium) [inline infra/base-images/base-builder/Dockerfile]: nit: this block can be shortened to
ENV GOPATH /root/go
ENV PATH $PATH:/root/.go/bin:$GOPATH/bin

Also, why do we have /root/go and /root/.go ?

File: infra/base-images/base-builder/Dockerfile [hunk 3]
```diff
@@ -27,6 +27,27 @@ RUN apt-get install -y git \
...
+ENV GOPATH /root/go
+ENV PATH $PATH:$GOPATH/bin
+
+# Dependency of go-fuzz
```
Comment (inferno-chromium) [inline infra/base-images/base-builder/Dockerfile]: nit: end with period.

**Comment by Dor1s** [comment]
Confirmed locally that this doesn't break golang project integration.

**Pull request merged**